Step3.5.md

```diff
@@ -0,0 +1,42 @@
+The below is actually from BARD :O
+
+Step 3.5: Front-End Deployment to S3 and Cloudfront
+Summary: Deploy the compiled front-end code to S3 as a static website served through Cloudfront.
+
+Description: This step involves building the front-end code using Vite, creating an S3 bucket for hosting the static files, configuring the bucket for static website hosting, and setting up a Cloudfront distribution for improved performance and global reach.
+
+Dependencies:
+
+Front-end code built with Vite
+AWS Account
+S3 Bucket
+Cloudfront Distribution
+Steps:
+
+Build the front-end code: Run the appropriate Vite command to build the production bundle. This typically involves a command like npm run build.
+Create an S3 bucket: In the AWS Console, create a new S3 bucket. Choose a unique and descriptive name for the bucket.
+Enable static website hosting: Enable static website hosting for the S3 bucket. Set the index document to index.html and the error document to index.html.
+Upload the built files: Upload the contents of the dist folder (generated by Vite) to the S3 bucket.
+Set access permissions: Configure the S3 bucket to allow public read access to the uploaded files.
+Create a Cloudfront distribution: Create a Cloudfront distribution with the S3 bucket as the origin. Choose the appropriate settings for your desired performance and security.
+Update DNS records: Update your DNS records to point the domain name to the Cloudfront distribution.
+Acceptance Criteria:
+
+The front-end code is compiled and bundled using Vite.
+An S3 bucket is created for static website hosting.
+Static website hosting is enabled on the S3 bucket.
+Built files are uploaded to the S3 bucket.
+Public read access is granted to the uploaded files.
+A Cloudfront distribution is created with the S3 bucket as the origin.
+DNS records are updated to point the domain name to the Cloudfront distribution.
+The front-end website is accessible through the domain name.
+Additional Notes:
+
+This step assumes basic familiarity with AWS S3 and Cloudfront services.
+Specific configurations for S3 and Cloudfront may vary depending on your specific needs.
+Consider implementing additional security measures like HTTPS and access control lists.
+Next Steps:
+
+Configure backend API integration with the deployed front-end application.
+Implement additional features and functionalities.
+Perform testing and quality assurance.
```

Comment:
should we ask bard how much this would cost approximately? Maybe also ask alternative approaches (just out of curiosity). The static site should be free

Reply:
We should keep it free tiered.
If we can use github workflows, if not we can just run this locally